src/start/AuthScreen.js

```diff
@@ -108,11 +120,30 @@ export const activeAuthentications = (
     result.push(auth);
   });
 
-  availableExternalMethods.forEach(auth => {
-    if (authenticationMethods[auth.name]) {
-      result.push(auth);
-    }
-  });
+  if (!externalAuthenticationMethods) {
+    // Server doesn't speak new API; get these methods from the old one.
+    availableExternalMethods.forEach(auth => {
+      if (authenticationMethods[auth.name]) {
+        result.push(auth);
+      }
+    });
+  } else {
+    // We have info from new API; ignore old one for these methods.
+    externalAuthenticationMethods.forEach(method => {
+      // The server provides icons as image URLs; but we have our own built
+      // in, which we don't have to load and can color to match the button.
+      // TODO perhaps switch to server's, for the sake of SAML where ours is
+      //   generic and the server may have a more specific one.
+      const Icon = externalMethodIcons.get(method.name) ?? IconPrivate;
```

Comment:
This is actually exactly the sort of thing that `name` is here for :slightly_smiling_face: . As the [docs](https://zulipchat.com/api/server-settings) put it:
>  [...] `name`, which is a unique, stable, machine-readable name for the authentication method.

The display name could easily get tweaked in the future, and the login URL certainly has been known to change (see comment about Google auth, above in this file). The `name` property is here so that when a client has some logic they want to apply to a particular authentication method, there's a reasonable test to hang it off of.